test/scenarios.spec.ts

```diff
@@ -454,3 +455,95 @@ suite('== Custom APL Props ==', () => {
         expect(dataSource).deep.equals(expectedDataSource);
     });
 });
+
+suite('== Custom Number APL Props ==', () => {
+    class NumberSelector extends ControlManager {
+        createControlTree(): Control {
+            const topControl = new ContainerControl({ id: 'root' });
+
+            // NumberControl
+            const numberControl = new NumberControl({
+                id: 'numItems',
+                validation: [
+                    (state) => state.value! % 2 === 0 || { renderedReason: 'the value must be even' },
+                ],
+                apl: {
+                    validationFailedMessage: 'The value must be even.',
+                    requestValue: {
+                        customHandlingFuncs: [{ canHandle: isNumberSelected, handle: handleNumberSelection }],
+                    },
+                },
+            });
+
+            function isNumberSelected(input: ControlInput) {
+                return InputUtil.isIntent(input, 'NumberSelectionIntent');
+            }
+
+            function handleNumberSelection(input: ControlInput, resultBuilder: ControlResultBuilder) {
+                if (getSupportedInterfaces(input.handlerInput.requestEnvelope)['Alexa.Presentation.APL']) {
+                    const intent = SimplifiedIntent.fromIntent((input.request as IntentRequest).intent);
+                    if (intent.slotResolutions.value !== undefined) {
+                        const listSelectedValue = intent.slotResolutions.value;
+                        numberControl.setValue(listSelectedValue.slotValue);
+                    }
+                }
+            }
+            topControl.addChild(numberControl);
+            return topControl;
+        }
+    }
+
+    test('APL custom handlers are invoked.', async () => {
+        // Note: this test demonstrates calling customHandlingFuncs if defined on a control
+
+        const rootControl = new NumberSelector().createControlTree();
+        const input = TestInput.of(
+            IntentBuilder.of('NumberSelectionIntent', {
+                value: '4',
+            }),
+        );
+        const result = new ControlResultBuilder(undefined!);
+        await rootControl.canHandle(input);
+        await rootControl.handle(input, result);
+        const numberControlState = findControlInTreeById(rootControl, 'numItems');
+        expect(numberControlState.state.value).eq(4);
+    });
+
+    test('APL dataSource sends custom validationFailedMessage.', async () => {
```

Comment:
please have a think about additional tests that we should have for GA.  we don't need to do everything right now, but if you have some in mind, pls add TODOs to the test files and we can look to add them to GA spec docs soon.

Soon we should write specification docs for each GA control like shreyas has done for MVSList... scenarios, UX decisions, high-level feature set and high-level test plan.